.github/workflows/push_dev_images.yml

```diff
@@ -0,0 +1,37 @@
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - 'main'
+      - 'releases/*'
```

Comment:
1) I really don't want to hardcode any logic to determine whether the image is or isn't up to date; 
    - this is redundant with functionality that is at the core of what docker provides, and 
    - this will change over time, so hardcoding it will likely lead to a broken pipeline in the future. 
2) I believe docker should be smart enough to recognize when layers are already available on the registry and merely have the new image/tag use those. See also the conversation here: https://forums.docker.com/t/why-does-docker-push-base-image-layers-that-are-already-on-docker-hub/2329/5 - I'd expect that should apply to any registry.

Reply:
1) I really don't want to hardcode any logic to determine whether the image is or isn't up to date; 
    - this is redundant with functionality that is at the core of what docker provides, and 
    - this will change over time, so hardcoding it will likely lead to a broken pipeline in the future. 
2) I believe docker should be smart enough to recognize when layers are already available on the registry and merely have the new image/tag use those. See also the conversation here: https://forums.docker.com/t/why-does-docker-push-base-image-layers-that-are-already-on-docker-hub/2329/5 and here https://github.com/docker/cli/issues/267#issuecomment-858736309.